android/java/org/chromium/chrome/browser/app/BraveActivity.java

```diff
@@ -1784,6 +1787,46 @@ public Profile getCurrentProfile() {
         return Profile.fromWebContents(tab.getWebContents());
     }
 
+    /** Close all tabs whose URL origin matches with a given origin. */
+    public void closeAllTabsByOrigin(@NonNull final String origin) {
+        final Tab activeTab = getActivityTab();
+        final TabModel tabModel = getCurrentTabModel();
+        if (activeTab != null && activeTab.getUrl().getOrigin().getSpec().equals(origin)) {
+            activeTab.setClosing(true);
+            tabModel.closeTab(activeTab);
+        }
+
+        Set<Integer> tabIndexes = TabModelUtils.getTabIndexesByUrlOrigin(tabModel, origin);
```

Comment:
Sadly, this class is not part of our codebase @supermassive. Any workaround to suggest?